src/main/java/uk/gov/digital/ho/hocs/info/api/TeamResource.java

```diff
@@ -148,6 +152,25 @@ public ResponseEntity deleteTeamPermissions(@PathVariable String teamUUID, @Requ
         return ResponseEntity.ok(teams.stream().map(TeamDto::from).collect(Collectors.toSet()));
     }
 
+    @GetMapping(value = "/team/users")
+    public ResponseEntity<Set<UserWithTeamsDto>> getUsersForTeam() {
+        List<UserDto> users = userService.getAllUsers();
+        Set<UserWithTeamsDto> usersWithTeams = new HashSet<>();
+
+        for (UserDto user: users){
+            Set<Team> teams = teamService.getTeamsForUser(UUID.fromString(user.getId()));
+            ArrayList<UUID> teamUUIDS = new ArrayList<>();
+            teams.forEach((team -> {
+                teamUUIDS.add(team.getUuid());
+                team.getUnit().getTeams().forEach((unitTeam) -> {
```

Comment:
Not sure I understand this - is this the case that a user is in team1 for unit1 and thus has all teams from unit1?

Reply:
This seemed to make sense to me, it's possible this is wrong, originally I tried to pass the teamDTO, but due to infinite recursion this broke the response, so I'm simply adding the teams and their units as this seemed to be more useful than just returning the team's uuid and asking the user to then retrieve the correct units on their own, I can change this to just returning the team's uuid quickly if that seems like a better idea

Reply:
If this shows what it should I'm happy to accept for now. Still reads slightly odd but I'll add it for something to look at.